src/core.ts

```diff
@@ -33,26 +33,29 @@ export class APIClient {
 
     let headers = {
       [API_KEY_HEADER]: this.apiKey,
-      ['content-type']: 'application/json',
       ...options.headers,
     };
-    if (body instanceof FormData) {
-      headers = {
-        ...headers,
-        ...body.getHeaders(),
-      };
+
+    // Bun will natively append headers like Content-Type, Content-Length, etc. so doesn't need to set them manually.
+    if (!isBunEnv() && body) {
+      if (body instanceof FormData) {
+        if (typeof (body as any).getHeaders === 'function') {
+          headers = {
+            ...headers,
+            ...(body as any).getHeaders(),
+          };
+        }
+      } else {
+        headers['content-type'] = 'application/json';
+        body = JSON.stringify(body);
+      }
```

Comment:
Double check for the condition:
we don't need to set header manually on Bun,
but `headers['content-type'] = 'application/json';` when isBunEnv()